all_channels_test.go

```diff
@@ -57,6 +62,7 @@ func TestAllChannelsRegistered(t *testing.T) {
 	ch1_1.Close()
 	ch2_1.Close()
 	ch2_2.Close()
+	time.Sleep(testutils.Timeout(10 * time.Millisecond))
```

Comment:
Since we moved `Channel.onClosed` to run async, we can't guarantee that `Close` immediately removes the channel from the introspection results.

Since this is the only place we're relying upon the current behavior, I opted to make these tests uglier instead of making the production code more complex. Alternatively, I can keep the call to `removeClosedChannel` in the `Close` method instead of putting it in `onClosed`, but that seems weird - introspection doesn't match reality.